launcher/container_runner.go

```diff
@@ -427,15 +409,45 @@ func (r *ContainerRunner) fetchAndWriteTokenWithRetry(ctx context.Context,
 	return nil
 }
 
-// defaultRetryPolicy retries with:
-// initial interval of 30s, multiplication factor of 1.5
-// randomization factor of 0.5, max interval of 3m, and
-// max elapsed time of 10m.
+func getNextRefreshFromExpiration(expiration time.Duration) time.Duration {
```

Comment:
Ideally, should have a unit test to test the returned time is always smaller than the expiration time.

Reply:
done. lmk if the test looks good to you.

Reply:
LGTM